ICE: (debug assertions) `suggestion must not have overlapping part`

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code

Needs rustc built with debug assertions!

```Rust
fn b() {
    b(
        Aa,
        a,
    )
}
```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
6d819a4b8f45b170e7c2c415df20cfa2e0cbbf7f
```

### Error output

```
error[E0425]: cannot find value `Aa` in this scope
 --> a.rs:3:9
  |
3 |         Aa,
  |         ^^ not found in this scope

error[E0425]: cannot find value `a` in this scope
 --> a.rs:4:9
  |
4 |         a,
  |         ^ not found in this scope

error[E0601]: `main` function not found in crate `a`
 --> a.rs:6:2
  |
6 | }
  |  ^ consider adding a `main` function to `a.rs`
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```

thread 'rustc' panicked at 'assertion failed: `(left == right)`
  left: `Some([SubstitutionPart { span: a.rs:3:9: 4:9 (#0), snippet: "" }, SubstitutionPart { span: a.rs:3:11: 4:10 (#0), snippet: "" }])`,
 right: `None`: suggestion must not have overlapping parts', /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:632:9
stack backtrace:
   0:     0x7f153e16ebf4 - std::backtrace_rs::backtrace::libunwind::trace::h1544161bb283d831
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f153e16ebf4 - std::backtrace_rs::backtrace::trace_unsynchronized::h88182f9ae043ea1d
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f153e16ebf4 - std::sys_common::backtrace::_print_fmt::hbc46e181236b160e
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x7f153e16ebf4 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h40f9d84c979d1c60
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f153e1da1b8 - core::fmt::write::h3693e9c93ce31d95
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/mod.rs:1232:17
   5:     0x7f153e15245f - std::io::Write::write_fmt::hf505aa7ad9258ee9
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/io/mod.rs:1684:15
   6:     0x7f153e16e9f5 - std::sys_common::backtrace::_print::h7f7e9448ce377511
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x7f153e16e9f5 - std::sys_common::backtrace::print::h10d8939f3c4a7090
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x7f153e135cf4 - std::panicking::default_hook::{{closure}}::hd96f1491f179fb0a
   9:     0x7f153e13595c - std::panicking::default_hook::hea690dedd054fe31
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:286:9
  10:     0x7f1540c69852 - <alloc[a96d60a9bf8c9b93]::boxed::Box<dyn for<'a, 'b> core[3a7799eb5b53fe42]::ops::function::Fn<(&'a core[3a7799eb5b53fe42]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[3a7799eb5b53fe42]::marker::Sync + core[3a7799eb5b53fe42]::marker::Send> as core[3a7799eb5b53fe42]::ops::function::Fn<(&core[3a7799eb5b53fe42]::panic::panic_info::PanicInfo,)>>::call
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:2002:9
  11:     0x7f1540c69852 - rustc_driver_impl[c7adae1617edfc64]::DEFAULT_HOOK::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:1145:17
  12:     0x7f153e1364d1 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hac989f8c5fbcacd1
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:2002:9
  13:     0x7f153e1364d1 - std::panicking::rust_panic_with_hook::hc778308fe861818c
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:692:13
  14:     0x7f153e16f229 - std::panicking::begin_panic_handler::{{closure}}::h18e0a9e49a0d5c86
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:579:13
  15:     0x7f153e16ecd6 - std::sys_common::backtrace::__rust_end_short_backtrace::h3cd1987cc33c3094
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:137:18
  16:     0x7f153e135ff2 - rust_begin_unwind
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:575:5
  17:     0x7f153e1eab33 - core::panicking::panic_fmt::ha9472d71642ed08e
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:64:14
  18:     0x7f153e1eb123 - core::panicking::assert_failed_inner::h08d1d4ab99760b32
  19:     0x7f15414ff279 - core[3a7799eb5b53fe42]::panicking::assert_failed::<core[3a7799eb5b53fe42]::option::Option<&[rustc_errors[e2c6533d414d421d]::SubstitutionPart; 2usize]>, core[3a7799eb5b53fe42]::option::Option<&[rustc_errors[e2c6533d414d421d]::SubstitutionPart; 2usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:211:5
  20:     0x7f15414aa764 - <rustc_errors[e2c6533d414d421d]::diagnostic::Diagnostic>::multipart_suggestion_with_style::<&alloc[a96d60a9bf8c9b93]::string::String>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:632:9
  21:     0x7f154129fd06 - <rustc_errors[e2c6533d414d421d]::diagnostic::Diagnostic>::multipart_suggestion::<&alloc[a96d60a9bf8c9b93]::string::String>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic.rs:588:9
  22:     0x7f154129fd06 - <rustc_errors[e2c6533d414d421d]::diagnostic_builder::DiagnosticBuilder<rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::multipart_suggestion::<&alloc[a96d60a9bf8c9b93]::string::String>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_errors/src/diagnostic_builder.rs:481:13
  23:     0x7f154129fd06 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::report_arg_errors
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1194:17
  24:     0x7f154129bcbc - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_argument_types
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:450:13
  25:     0x7f1541264ff2 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::confirm_builtin_call
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/callee.rs:450:9
  26:     0x7f1541262e75 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_call
  27:     0x7f15412f3663 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:329:45
  28:     0x7f154127ecb1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:235:18
  29:     0x7f154127ecb1 - stacker[12027ce31fbb92f0]::maybe_grow::<rustc_middle[d07bed17621820c2]::ty::Ty, <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  30:     0x7f154127ecb1 - rustc_data_structures[de9c3b5be789562e]::stack::ensure_sufficient_stack::<rustc_middle[d07bed17621820c2]::ty::Ty, <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  31:     0x7f154127ecb1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:231:18
  32:     0x7f15412f2a65 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:185:9
  33:     0x7f15412a41c1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1530:50
  34:     0x7f15412a41c1 - <core[3a7799eb5b53fe42]::option::Option<&&rustc_hir[4d49ff8807738c8c]::hir::Expr>>::map::<rustc_middle[d07bed17621820c2]::ty::Ty, <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/option.rs:972:29
  35:     0x7f15412a41c1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1530:32
  36:     0x7f15412a41c1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::with_breakable_ctxt::<<rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_block_with_expected::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs:1439:22
  37:     0x7f15412a41c1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_block_with_expected
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1523:26
  38:     0x7f15412f39d9 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:328:41
  39:     0x7f154127ecb1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:235:18
  40:     0x7f154127ecb1 - stacker[12027ce31fbb92f0]::maybe_grow::<rustc_middle[d07bed17621820c2]::ty::Ty, <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  41:     0x7f154127ecb1 - rustc_data_structures[de9c3b5be789562e]::stack::ensure_sufficient_stack::<rustc_middle[d07bed17621820c2]::ty::Ty, <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  42:     0x7f154127ecb1 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:231:18
  43:     0x7f15412f2a65 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:185:9
  44:     0x7f1541280aa8 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_expr_with_hint
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:137:9
  45:     0x7f1541280aa8 - <rustc_hir_typeck[81f0364ed15c14f6]::fn_ctxt::FnCtxt>::check_return_expr
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/expr.rs:835:30
  46:     0x7f1541470f63 - rustc_hir_typeck[81f0364ed15c14f6]::check::check_fn
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/check.rs:122:9
  47:     0x7f154149bb89 - rustc_hir_typeck[81f0364ed15c14f6]::typeck_with_fallback::<rustc_hir_typeck[81f0364ed15c14f6]::typeck::{closure#0}>::{closure#0}::{closure#1}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:219:13
  48:     0x7f154149bb89 - <rustc_hir_typeck[81f0364ed15c14f6]::inherited::InheritedBuilder>::enter::<rustc_hir_typeck[81f0364ed15c14f6]::typeck_with_fallback<rustc_hir_typeck[81f0364ed15c14f6]::typeck::{closure#0}>::{closure#0}::{closure#1}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/inherited.rs:108:9
  49:     0x7f154149bb89 - rustc_hir_typeck[81f0364ed15c14f6]::typeck_with_fallback::<rustc_hir_typeck[81f0364ed15c14f6]::typeck::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:202:26
  50:     0x7f154149bb89 - rustc_hir_typeck[81f0364ed15c14f6]::typeck_with_fallback::<rustc_hir_typeck[81f0364ed15c14f6]::typeck::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:185:36
  51:     0x7f154149bb89 - rustc_hir_typeck[81f0364ed15c14f6]::typeck
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:166:9
  52:     0x7f154287f980 - <rustc_query_impl[bf56cdb236176669]::queries::typeck as rustc_query_system[c5079e9cea6221c8]::query::config::QueryConfig<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:507:21
  53:     0x7f154287f980 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job::<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:425:71
  54:     0x7f154287f980 - stacker[12027ce31fbb92f0]::maybe_grow::<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  55:     0x7f154287f980 - rustc_data_structures[de9c3b5be789562e]::stack::ensure_sufficient_stack::<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  56:     0x7f154287f980 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:128:17
  57:     0x7f154287f980 - rustc_middle[d07bed17621820c2]::ty::context::tls::tlv::with_tlv::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
  58:     0x7f154287f980 - rustc_middle[d07bed17621820c2]::ty::context::tls::enter_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
  59:     0x7f154287f980 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:127:13
  60:     0x7f154287f980 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:164:9
  61:     0x7f154287f980 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:36
  62:     0x7f154287f980 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context_opt::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_context<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:131:18
  63:     0x7f154287f980 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:5
  64:     0x7f154287f980 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, &rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:155:5
  65:     0x7f154287f980 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<&rustc_middle[d07bed17621820c2]::ty::typeck_results::TypeckResults, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:112:9
  66:     0x7f154287f980 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job::<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:425:22
  67:     0x7f154287f980 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::try_execute_query::<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:365:17
  68:     0x7f1542cc0763 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::get_query::<rustc_query_impl[bf56cdb236176669]::queries::typeck, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt, rustc_middle[d07bed17621820c2]::dep_graph::dep_node::DepKind>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:739:36
  69:     0x7f1542cc0763 - <rustc_query_impl[bf56cdb236176669]::Queries as rustc_middle[d07bed17621820c2]::ty::query::QueryEngine>::typeck::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:744:17
  70:     0x7f1542cc0763 - <rustc_query_impl[bf56cdb236176669]::Queries as rustc_middle[d07bed17621820c2]::ty::query::QueryEngine>::typeck
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:46:1
  71:     0x7f15413b291c - <rustc_middle[d07bed17621820c2]::ty::query::TyCtxtEnsure>::typeck::<rustc_span[82580b446b48c745]::def_id::LocalDefId>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:284:29
  72:     0x7f15413b291c - rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:150:64
  73:     0x7f15413b291c - <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners::<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/hir/map/mod.rs:491:82
  74:     0x7f15413b291c - rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in::<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:152:66
  75:     0x7f15413b291c - <core[3a7799eb5b53fe42]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}> as core[3a7799eb5b53fe42]::ops::function::FnOnce<()>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panic/unwind_safe.rs:271:9
  76:     0x7f15413b291c - std[6ee66c0b76df7c3f]::panicking::try::do_call::<core[3a7799eb5b53fe42]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:483:40
  77:     0x7f15413b291c - std[6ee66c0b76df7c3f]::panicking::try::<(), core[3a7799eb5b53fe42]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:447:19
  78:     0x7f15413b291c - std[6ee66c0b76df7c3f]::panic::catch_unwind::<core[3a7799eb5b53fe42]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}::{closure#0}>, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panic.rs:140:14
  79:     0x7f15413b291c - rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in::<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:152:33
  80:     0x7f15413b291c - <core[3a7799eb5b53fe42]::slice::iter::Iter<rustc_span[82580b446b48c745]::def_id::LocalDefId> as core[3a7799eb5b53fe42]::iter::traits::iterator::Iterator>::for_each::<rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/slice/iter/macros.rs:201:21
  81:     0x7f15413b291c - rustc_data_structures[de9c3b5be789562e]::sync::par_for_each_in::<&[rustc_span[82580b446b48c745]::def_id::LocalDefId], <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/sync.rs:151:13
  82:     0x7f1541498b60 - <rustc_middle[d07bed17621820c2]::hir::map::Map>::par_body_owners::<rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/hir/map/mod.rs:491:9
  83:     0x7f1541498b60 - rustc_hir_typeck[81f0364ed15c14f6]::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_typeck/src/lib.rs:150:5
  84:     0x7f15427e6bf4 - <rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies as rustc_query_system[c5079e9cea6221c8]::query::config::QueryConfig<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:507:21
  85:     0x7f15427e6bf4 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job::<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:425:71
  86:     0x7f15427e6bf4 - stacker[12027ce31fbb92f0]::maybe_grow::<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
  87:     0x7f15427e6bf4 - rustc_data_structures[de9c3b5be789562e]::stack::ensure_sufficient_stack::<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  88:     0x7f15427e6bf4 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:128:17
  89:     0x7f15427e6bf4 - rustc_middle[d07bed17621820c2]::ty::context::tls::tlv::with_tlv::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
  90:     0x7f15427e6bf4 - rustc_middle[d07bed17621820c2]::ty::context::tls::enter_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
  91:     0x7f15427e6bf4 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:127:13
  92:     0x7f15427e6bf4 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:164:9
  93:     0x7f15427e6bf4 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:36
  94:     0x7f15427e6bf4 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context_opt::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_context<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:131:18
  95:     0x7f15427e6bf4 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:5
  96:     0x7f15427e6bf4 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, ()>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:155:5
  97:     0x7f15427e6bf4 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<(), rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:112:9
  98:     0x7f15427e6bf4 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job::<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:425:22
  99:     0x7f15427e6bf4 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::try_execute_query::<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:365:17
 100:     0x7f1542cc0041 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::get_query::<rustc_query_impl[bf56cdb236176669]::queries::typeck_item_bodies, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt, rustc_middle[d07bed17621820c2]::dep_graph::dep_node::DepKind>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:739:36
 101:     0x7f1542cc0041 - <rustc_query_impl[bf56cdb236176669]::Queries as rustc_middle[d07bed17621820c2]::ty::query::QueryEngine>::typeck_item_bodies::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:744:17
 102:     0x7f1542cc0041 - <rustc_query_impl[bf56cdb236176669]::Queries as rustc_middle[d07bed17621820c2]::ty::query::QueryEngine>::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:46:1
 103:     0x7f1540fa6f26 - <rustc_middle[d07bed17621820c2]::ty::query::TyCtxtAt>::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:309:29
 104:     0x7f1540fa6f26 - <rustc_middle[d07bed17621820c2]::ty::context::TyCtxt>::typeck_item_bodies
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:295:35
 105:     0x7f1540fa6f26 - rustc_hir_analysis[b8ce6fedbd05fcc0]::check_crate::{closure#7}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/lib.rs:531:50
 106:     0x7f1540fa6f26 - <rustc_data_structures[de9c3b5be789562e]::profiling::VerboseTimingGuard>::run::<(), rustc_hir_analysis[b8ce6fedbd05fcc0]::check_crate::{closure#7}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/profiling.rs:723:9
 107:     0x7f1540fa6f26 - <rustc_session[ec4229a09ed780b3]::session::Session>::time::<(), rustc_hir_analysis[b8ce6fedbd05fcc0]::check_crate::{closure#7}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_session/src/utils.rs:10:50
 108:     0x7f154110a075 - rustc_hir_analysis[b8ce6fedbd05fcc0]::check_crate
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/lib.rs:531:5
 109:     0x7f1541852d67 - rustc_interface[22044caa1f29908d]::passes::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_interface/src/passes.rs:745:5
 110:     0x7f1542882a86 - <rustc_query_impl[bf56cdb236176669]::queries::analysis as rustc_query_system[c5079e9cea6221c8]::query::config::QueryConfig<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:507:21
 111:     0x7f1542882a86 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job::<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:425:71
 112:     0x7f1542882a86 - stacker[12027ce31fbb92f0]::maybe_grow::<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/github.com-1ecc6299db9ec823/stacker-0.1.15/src/lib.rs:55:9
 113:     0x7f1542882a86 - rustc_data_structures[de9c3b5be789562e]::stack::ensure_sufficient_stack::<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
 114:     0x7f1542882a86 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:128:17
 115:     0x7f1542882a86 - rustc_middle[d07bed17621820c2]::ty::context::tls::tlv::with_tlv::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
 116:     0x7f1542882a86 - rustc_middle[d07bed17621820c2]::ty::context::tls::enter_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
 117:     0x7f1542882a86 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:127:13
 118:     0x7f1542882a86 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:164:9
 119:     0x7f1542882a86 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:36
 120:     0x7f1542882a86 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context_opt::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_context<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:131:18
 121:     0x7f1542882a86 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_context::<rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:142:5
 122:     0x7f1542882a86 - rustc_middle[d07bed17621820c2]::ty::context::tls::with_related_context::<<rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:155:5
 123:     0x7f1542882a86 - <rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt as rustc_query_system[c5079e9cea6221c8]::query::QueryContext>::start_query::<core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>, rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:112:9
 124:     0x7f1542882a86 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::execute_job::<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:425:22
 125:     0x7f1542882a86 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::try_execute_query::<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:365:17
 126:     0x7f1542c98a27 - rustc_query_system[c5079e9cea6221c8]::query::plumbing::get_query::<rustc_query_impl[bf56cdb236176669]::queries::analysis, rustc_query_impl[bf56cdb236176669]::plumbing::QueryCtxt, rustc_middle[d07bed17621820c2]::dep_graph::dep_node::DepKind>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:739:36
 127:     0x7f1542c98a27 - <rustc_query_impl[bf56cdb236176669]::Queries as rustc_middle[d07bed17621820c2]::ty::query::QueryEngine>::analysis::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:744:17
 128:     0x7f1542c98a27 - <rustc_query_impl[bf56cdb236176669]::Queries as rustc_middle[d07bed17621820c2]::ty::query::QueryEngine>::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:46:1
 129:     0x7f1540c6bc9c - <rustc_middle[d07bed17621820c2]::ty::query::TyCtxtAt>::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:309:29
 130:     0x7f1540c6bc9c - <rustc_middle[d07bed17621820c2]::ty::context::TyCtxt>::analysis
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/query.rs:295:35
 131:     0x7f1540c6bc9c - rustc_driver_impl[c7adae1617edfc64]::run_compiler::{closure#1}::{closure#2}::{closure#3}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:343:52
 132:     0x7f1540c6bc9c - <rustc_middle[d07bed17621820c2]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[c7adae1617edfc64]::run_compiler::{closure#1}::{closure#2}::{closure#3}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context.rs:554:37
 133:     0x7f1540c6bc9c - rustc_middle[d07bed17621820c2]::ty::context::tls::tlv::with_tlv::<<rustc_middle[d07bed17621820c2]::ty::context::GlobalCtxt>::enter<rustc_driver_impl[c7adae1617edfc64]::run_compiler::{closure#1}::{closure#2}::{closure#3}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:94:9
 134:     0x7f1540c6bc9c - rustc_middle[d07bed17621820c2]::ty::context::tls::enter_context::<<rustc_middle[d07bed17621820c2]::ty::context::GlobalCtxt>::enter<rustc_driver_impl[c7adae1617edfc64]::run_compiler::{closure#1}::{closure#2}::{closure#3}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>::{closure#0}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:114:5
 135:     0x7f1540c6bc9c - <rustc_middle[d07bed17621820c2]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[c7adae1617edfc64]::run_compiler::{closure#1}::{closure#2}::{closure#3}, core[3a7799eb5b53fe42]::result::Result<(), rustc_errors[e2c6533d414d421d]::ErrorGuaranteed>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context.rs:554:9
 
<snip>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.69.0-dev running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [typeck] type-checking `b`
#1 [typeck_item_bodies] type-checking all item bodies
#2 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 3 previous errors

Some errors have detailed explanations: E0425, E0601.
For more information about an error, try `rustc --explain E0425`.
```

</p>
</details>

--- a/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs
+++ b/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs
@@ -932,25 +932,22 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     labels
                         .push((provided_span, format!("unexpected argument{}", provided_ty_name)));
                     let mut span = provided_span;
-                    if arg_idx.index() > 0
+                    if span.can_be_used_for_suggestions() {
+                        if arg_idx.index() > 0
                         && let Some((_, prev)) = provided_arg_tys
                             .get(ProvidedIdx::from_usize(arg_idx.index() - 1)
                     ) {
                         // Include previous comma
-                        span = span.with_lo(prev.hi());
-                    } else if let Some((_, next)) = provided_arg_tys.get(
-                        ProvidedIdx::from_usize(arg_idx.index() + 1),
-                    ) {
-                        // Include next comma
-                        span = span.until(*next);
+                        span = prev.shrink_to_hi().to(span);
                     }
-                    suggestions.push((span, String::new()));
+                        suggestions.push((span, String::new()));
 
-                    suggestion_text = match suggestion_text {
-                        SuggestionText::None => SuggestionText::Remove(false),
-                        SuggestionText::Remove(_) => SuggestionText::Remove(true),
-                        _ => SuggestionText::DidYouMean,
-                    };
+                        suggestion_text = match suggestion_text {
+                            SuggestionText::None => SuggestionText::Remove(false),
+                            SuggestionText::Remove(_) => SuggestionText::Remove(true),
+                            _ => SuggestionText::DidYouMean,
+                        };
+                    }
                 }
                 Error::Missing(expected_idx) => {
                     // If there are multiple missing arguments adjacent to each other,

--- a/tests/ui/alloc-error/alloc-error-handler-bad-signature-3.stderr
+++ b/tests/ui/alloc-error/alloc-error-handler-bad-signature-3.stderr
@@ -7,10 +7,7 @@ LL |   fn oom() -> ! {
    |  _-^^^^^^^^^^^^
 LL | |     loop {}
 LL | | }
-   | | -
-   | | |
-   | |_unexpected argument of type `core::alloc::Layout`
-   |   help: remove the extra argument
+   | |_- unexpected argument of type `core::alloc::Layout`
    |
 note: function defined here
   --> $DIR/alloc-error-handler-bad-signature-3.rs:10:4

--- a/tests/ui/argument-suggestions/extra_arguments.rs
+++ b/tests/ui/argument-suggestions/extra_arguments.rs
@@ -3,8 +3,15 @@ fn one_arg(_a: i32) {}
 fn two_arg_same(_a: i32, _b: i32) {}
 fn two_arg_diff(_a: i32, _b: &str) {}
 
+macro_rules! foo {
+    ($x:expr) => {
+        empty($x, 1); //~ ERROR function takes
+    }
+}
+
 fn main() {
   empty(""); //~ ERROR function takes
+  empty(1, 1); //~ ERROR function takes
 
   one_arg(1, 1); //~ ERROR function takes
   one_arg(1, ""); //~ ERROR function takes
@@ -32,4 +39,5 @@ fn main() {
     1,
     ""
   );
+  foo!(1);
 }

--- a/tests/ui/argument-suggestions/extra_arguments.stderr
+++ b/tests/ui/argument-suggestions/extra_arguments.stderr
@@ -1,5 +1,5 @@
 error[E0061]: this function takes 0 arguments but 1 argument was supplied
-  --> $DIR/extra_arguments.rs:7:3
+  --> $DIR/extra_arguments.rs:13:3
    |
 LL |   empty("");
    |   ^^^^^ --
@@ -13,8 +13,27 @@ note: function defined here
 LL | fn empty() {}
    |    ^^^^^
 
+error[E0061]: this function takes 0 arguments but 2 arguments were supplied
+  --> $DIR/extra_arguments.rs:14:3
+   |
+LL |   empty(1, 1);
+   |   ^^^^^ -  - unexpected argument of type `{integer}`
+   |         |
+   |         unexpected argument of type `{integer}`
+   |
+note: function defined here
+  --> $DIR/extra_arguments.rs:1:4
+   |
+LL | fn empty() {}
+   |    ^^^^^
+help: remove the extra arguments
+   |
+LL -   empty(1, 1);
+LL +   empty();
+   |
+
 error[E0061]: this function takes 1 argument but 2 arguments were supplied
-  --> $DIR/extra_arguments.rs:9:3
+  --> $DIR/extra_arguments.rs:16:3
    |
 LL |   one_arg(1, 1);
    |   ^^^^^^^  ---
@@ -29,7 +48,7 @@ LL | fn one_arg(_a: i32) {}
    |    ^^^^^^^ -------
 
 error[E0061]: this function takes 1 argument but 2 arguments were supplied
-  --> $DIR/extra_arguments.rs:10:3
+  --> $DIR/extra_arguments.rs:17:3
    |
 LL |   one_arg(1, "");
    |   ^^^^^^^  ----
@@ -44,7 +63,7 @@ LL | fn one_arg(_a: i32) {}
    |    ^^^^^^^ -------
 
 error[E0061]: this function takes 1 argument but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:11:3
+  --> $DIR/extra_arguments.rs:18:3
    |
 LL |   one_arg(1, "", 1.0);
    |   ^^^^^^^    --  --- unexpected argument of type `{float}`
@@ -63,7 +82,7 @@ LL +   one_arg(1);
    |
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:13:3
+  --> $DIR/extra_arguments.rs:20:3
    |
 LL |   two_arg_same(1, 1, 1);
    |   ^^^^^^^^^^^^     ---
@@ -78,7 +97,7 @@ LL | fn two_arg_same(_a: i32, _b: i32) {}
    |    ^^^^^^^^^^^^ -------  -------
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:14:3
+  --> $DIR/extra_arguments.rs:21:3
    |
 LL |   two_arg_same(1, 1, 1.0);
    |   ^^^^^^^^^^^^     -----
@@ -93,7 +112,7 @@ LL | fn two_arg_same(_a: i32, _b: i32) {}
    |    ^^^^^^^^^^^^ -------  -------
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:16:3
+  --> $DIR/extra_arguments.rs:23:3
    |
 LL |   two_arg_diff(1, 1, "");
    |   ^^^^^^^^^^^^  ---
@@ -108,7 +127,7 @@ LL | fn two_arg_diff(_a: i32, _b: &str) {}
    |    ^^^^^^^^^^^^ -------  --------
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:17:3
+  --> $DIR/extra_arguments.rs:24:3
    |
 LL |   two_arg_diff(1, "", "");
    |   ^^^^^^^^^^^^      ----
@@ -123,7 +142,7 @@ LL | fn two_arg_diff(_a: i32, _b: &str) {}
    |    ^^^^^^^^^^^^ -------  --------
 
 error[E0061]: this function takes 2 arguments but 4 arguments were supplied
-  --> $DIR/extra_arguments.rs:18:3
+  --> $DIR/extra_arguments.rs:25:3
    |
 LL |   two_arg_diff(1, 1, "", "");
    |   ^^^^^^^^^^^^    -      -- unexpected argument of type `&'static str`
@@ -142,7 +161,7 @@ LL +   two_arg_diff(1, "");
    |
 
 error[E0061]: this function takes 2 arguments but 4 arguments were supplied
-  --> $DIR/extra_arguments.rs:19:3
+  --> $DIR/extra_arguments.rs:26:3
    |
 LL |   two_arg_diff(1, "", 1, "");
    |   ^^^^^^^^^^^^        -  -- unexpected argument of type `&'static str`
@@ -161,7 +180,7 @@ LL +   two_arg_diff(1, "");
    |
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:22:3
+  --> $DIR/extra_arguments.rs:29:3
    |
 LL |   two_arg_same(1, 1,     "");
    |   ^^^^^^^^^^^^     --------
@@ -176,7 +195,7 @@ LL | fn two_arg_same(_a: i32, _b: i32) {}
    |    ^^^^^^^^^^^^ -------  -------
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:23:3
+  --> $DIR/extra_arguments.rs:30:3
    |
 LL |   two_arg_diff(1, 1,     "");
    |   ^^^^^^^^^^^^  ---
@@ -191,7 +210,7 @@ LL | fn two_arg_diff(_a: i32, _b: &str) {}
    |    ^^^^^^^^^^^^ -------  --------
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:24:3
+  --> $DIR/extra_arguments.rs:31:3
    |
 LL |     two_arg_same(
    |     ^^^^^^^^^^^^
@@ -211,7 +230,7 @@ LL | fn two_arg_same(_a: i32, _b: i32) {}
    |    ^^^^^^^^^^^^ -------  -------
 
 error[E0061]: this function takes 2 arguments but 3 arguments were supplied
-  --> $DIR/extra_arguments.rs:30:3
+  --> $DIR/extra_arguments.rs:37:3
    |
 LL |     two_arg_diff(
    |     ^^^^^^^^^^^^
@@ -229,6 +248,26 @@ note: function defined here
 LL | fn two_arg_diff(_a: i32, _b: &str) {}
    |    ^^^^^^^^^^^^ -------  --------
 
-error: aborting due to 14 previous errors
+error[E0061]: this function takes 0 arguments but 2 arguments were supplied
+  --> $DIR/extra_arguments.rs:8:9
+   |
+LL |         empty($x, 1);
+   |         ^^^^^     - unexpected argument of type `{integer}`
+...
+LL |   foo!(1);
+   |   -------
+   |   |    |
+   |   |    unexpected argument of type `{integer}`
+   |   |    help: remove the extra argument
+   |   in this macro invocation
+   |
+note: function defined here
+  --> $DIR/extra_arguments.rs:1:4
+   |
+LL | fn empty() {}
+   |    ^^^^^
+   = note: this error originates in the macro `foo` (in Nightly builds, run with -Z macro-backtrace for more info)
+
+error: aborting due to 16 previous errors
 
 For more information about this error, try `rustc --explain E0061`.

--- a/tests/ui/issues/issue-26094.rs
+++ b/tests/ui/issues/issue-26094.rs
@@ -1,7 +1,7 @@
 macro_rules! some_macro {
-    ($other: expr) => ({
+    ($other: expr) => {{
         $other(None) //~ NOTE unexpected argument of type `Option<_>`
-    })
+    }};
 }
 
 fn some_function() {} //~ NOTE defined here
@@ -9,5 +9,4 @@ fn some_function() {} //~ NOTE defined here
 fn main() {
     some_macro!(some_function);
     //~^ ERROR function takes 0 arguments but 1 argument was supplied
-    //~| NOTE in this expansion of some_macro!
 }

--- a/tests/ui/issues/issue-26094.stderr
+++ b/tests/ui/issues/issue-26094.stderr
@@ -2,10 +2,7 @@ error[E0061]: this function takes 0 arguments but 1 argument was supplied
   --> $DIR/issue-26094.rs:10:17
    |
 LL |         $other(None)
-   |                ----
-   |                |
-   |                unexpected argument of type `Option<_>`
-   |                help: remove the extra argument
+   |                ---- unexpected argument of type `Option<_>`
 ...
 LL |     some_macro!(some_function);
    |                 ^^^^^^^^^^^^^

